pandas/io/json/_normalize.py

```diff
@@ -228,13 +228,17 @@ def json_normalize(
     Returns normalized data with columns prefixed with the given string.
     """
 
-    def _pull_field(js, spec):
-        result = js
+    def _pull_field(js: Dict[str, Any], spec: Union[List, str]) -> List:
         if isinstance(spec, list):
+            _res = js
```

Comment:
What is the point of this assignment here? I think without this don't need all of the mypy discussion below

Reply:
@WillAyd Actually, the opposite was true. I set `result: List` which made it incompatible with `result = result[field]` (`result` was set to `js` above. `js` is not List and neither always iterable)

Reply:
What's the point of assigning this here and then overwriting it on line 235 within the loop?

Reply:
I just tested your solution. Tests do not pass. A `TypeError` is thrown.

Reply:
@WillAyd the loop drills down into the js. `_res` can be anything, `result` cannot be. `js[field]` can be anything hence I cannot assign `result` to `js[field]`

Reply:
Is it a TypeError on line 236? Why is the rhs of that referencing `_res` instead of `js`?

Reply:
It is a loop that goes over a nested `js`. Hence it reassigns it to do the drill down according to `spec`. 

```
    def _pull_field(js: Dict[str, Any], spec: Union[List, str]) -> Iterable:
        result: Any = []
        if isinstance(spec, list):
            for field in spec:
>               result = _result[field]
E               TypeError: list indices must be integers or slices, not str
```

Reply:
I see that I made a mistake in my code to test it (I never did `result = js`)

Reply:
Let me quickly check that

Reply:
@WillAyd apologies, your sugestion does work. PR updated.